examples/dummy-iiostream.c

```diff
@@ -227,13 +227,14 @@ int main (int argc, char **argv)
 	// Listen to ctrl+c and assert
 	signal(SIGINT, handle_sig);
 
-	unsigned int major, minor;
+	unsigned int i, j, major, minor;
 	char git_tag[8];
 	iio_library_get_version(&major, &minor, git_tag);
 	printf("Library version: %u.%u (git tag: %s)\n", major, minor, git_tag);
 
-	/* check for struct iio_data_format.repeat support */
-	has_repeat = major >= 0 && minor >= 8 ? true : false;
+	/* check for struct iio_data_format.repeat support
+	 * 0.8 has repeat support, don't need to check major, it's unsigned */
+	has_repeat = minor >= 8 ? true : false;
```

Comment:
this looks problematic when we bump version to 1.0;
we may want to make a note to do a `LIBIIO_VERSION(major, minor)    ((major << 8) | minor)` macro or something like that;
or maybe a helper function called  `bool iio_library_version_is_at_least(major, minor)`

obviously, this can be a separate patch;

Reply:
maybe a 2nd thought: is it worth doing this check? we don't recommend people use 0.8 anymore;
so, is it a good time to remove it now? [for such old versions]

Reply:
I fixed this. 

0.8 was released on Nov 4, 2016; so that was awhile ago, since it's all over in the code (has_repeat) - untangling things is a pain.